ledger-api/grpc-definitions/com/daml/ledger/api/v1/connected_domains_service.proto

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1;
+
+option java_outer_classname = "ConnectedDomainsServiceOuterClass";
+option java_package = "com.daml.ledger.api.v1";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1";
+
+// Allows clients to get state of connected domains from the ledger.
+service ConnectedDomainsService {
+  // Get the list of connected domains.
+  rpc GetConnectedDomains (GetConnectedDomainsRequest) returns (GetConnectedDomainsResponse);
+}
+
+message GetConnectedDomainsRequest {
+  // The submitting party
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string party = 1;
+}
+
+message GetConnectedDomainsResponse {
+  message ConnectedDomain {
+    // The alias of the domain
+    // Required
+    string domain_alias = 1;
+    // The ID of the domain
+    // Required
+    string domain_id = 2;
+    // The permission on the domain
+    // Required
+    ParticipantPermission permission = 3;
+  }
+  repeated ConnectedDomain connected_domains = 1;
+}
+
+// enum indicating the permission level
+enum ParticipantPermission {
+  MissingParticipantPermission = 0;
```

Comment:
can I ever get this?

Reply:
good question, @rgugliel-da, @mlachkar-da  do you know maybe?

Reply:
Good question. Asked in #team-canton, let's see :)

Reply:
Answer from Ratko:
"I was forced at gun-point by my more strict fellow engineers to explicitly add the 0 value of an enum as “missing” to prevent the situation where the absence of a value is mapped to the default value.
so this is only on the protobuf layer deserialization path. not on the serialization path."

Reply:
As discussed we will stick to the same on the API for symmetry. Is it okay with you @cocreature ?
From the code it seems to me you can never get that.

Reply:
Removed the missing- and shifted the indexes as discussed